code/game/objects/items/floppy_disk.dm

```diff
@@ -0,0 +1,346 @@
+		return CLICK_ACTION_BLOCKING
+
+	var/list/items = list()
+	for(var/variant_name in icon_variants)
+		var/icon_state_name = icon_variants[variant_name]
+		var/image/item_image = image(icon = icon, icon_state = icon_state)
+
+		var/image/overlay_preview = image(icon = icon, icon_state = icon_state_name)
+		item_image.overlays += overlay_preview
+		items += list("[variant_name]" = item_image)
+
+	var/pick = show_radial_menu(user, src, items, custom_check = CALLBACK(src, PROC_REF(check_sticker_menu), user), radius = 38, require_near = TRUE)
+	if(!pick)
+		return CLICK_ACTION_BLOCKING
+	if(!icon_variants[pick])
+		return CLICK_ACTION_BLOCKING
+
+	cut_overlays()
+	add_overlay(icon_variants[pick])
+	initial_sticker_icon_state = icon_variants[pick]
+	sticker_changed = TRUE
+	to_chat(user, span_notice("You change the sticker on [src] to '[pick]'."))
+	return CLICK_ACTION_SUCCESS
+
+/obj/item/disk/proc/check_sticker_menu(mob/user)
+	if(QDELETED(src))
+		return FALSE
+	if(sticker_changed)
+		return FALSE
+	if(!istype(user))
+		return FALSE
+	if(user.incapacitated)
+		return FALSE
+	return TRUE
+
+/obj/item/disk_stack
+	name = "stack of floppy disks"
+	desc = "A stack of floppy disks. You wonder what happens if you pull out the bottom one..."
+	icon = null
+	icon_state = null
+	w_class = WEIGHT_CLASS_SMALL
+	inhand_icon_state = "card-id"
+	lefthand_file = 'icons/mob/inhands/equipment/idcards_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/idcards_righthand.dmi'
+	drop_sound = 'sound/items/handling/disk_drop.ogg'
+	pickup_sound = 'sound/items/handling/disk_pickup.ogg'
+	var/list/stacked_disks = list()
+
+/obj/item/disk_stack/Initialize(mapload)
+	. = ..()
+	RegisterSignal(src, COMSIG_MOVABLE_THROW_LANDED, PROC_REF(spread))
+
+/obj/item/disk_stack/examine(mob/user)
+	. = ..()
+	. += span_notice("There are [span_bold("[length(stacked_disks)]")] disks in the stack.")
+
+/obj/item/disk/proc/handle_interaction(mob/living/user, obj/item/other)
+	if(istype(other, /obj/item/disk_stack))
+		var/obj/item/disk_stack/held_stack = other
+		var/obj/item/disk_stack/new_stack = new(get_turf(src))
+		var/was_in_hand = user.is_holding(src) || user.is_holding(held_stack)
+		new_stack.add_to_stack(user, src)
+
+		for(var/obj/item/disk/disk_from_hand in held_stack.stacked_disks)
+			if(length(new_stack.stacked_disks) >= MAX_DISK_STACK_SIZE)
+				break
+
+			disk_from_hand.forceMove(new_stack)
+			new_stack.stacked_disks += disk_from_hand
+
+		held_stack.stacked_disks.Cut()
+		new_stack.update_overlays()
+		if(was_in_hand)
+			user.put_in_hands(new_stack)
+		QDEL_IN(held_stack, 0)
+
+		balloon_alert(user, "merged stacks")
+		return ITEM_INTERACT_SUCCESS
+
+	if(istype(other, /obj/item/disk))
+		var/obj/item/disk_stack/new_stack = new(get_turf(src))
+		new_stack.pixel_x = pixel_x
+		new_stack.pixel_y = pixel_y
+
+		var/was_in_hand = user.is_holding(src)
+		new_stack.add_to_stack(user, src)
+		new_stack.add_to_stack(user, other)
+		new_stack.update_overlays()
```

Comment:
should be calling `update_appearance(UPDATE_OVERLAYS)` instead of calling `update_overlays()` directly